calyx/src/passes/dead_cell_removal.rs

```diff
@@ -52,24 +52,23 @@ impl Visitor for DeadCellRemoval {
         _sigs: &LibrarySignatures,
     ) -> VisResult {
         // All cells used in groups
-        for group in &comp.groups {
+        for group in comp.iter_groups() {
             self.used_cells.extend(
                 &mut analysis::ReadWriteSet::uses(&group.borrow().assignments)
                     .into_iter()
-                    .map(|c| c.borrow().name.clone()),
+                    .map(|c| c.borrow().name().clone()),
             )
         }
 
         // All cells used in continuous assignments.
         self.used_cells.extend(
             &mut analysis::ReadWriteSet::uses(&comp.continuous_assignments)
                 .into_iter()
-                .map(|c| c.borrow().name.clone()),
+                .map(|c| c.borrow().name().clone()),
```

Comment:
I wonder if we can have a `GetName` trait that we implement for `RRC<Group>` and `RRC<Component>` etc. so that we don't have to write `.borrow()` everywhere. This is just a thought and maybe not part of this PR.

Reply:
I was having a similar thought while writing this, could be nice. The only downside is covering up the borrow could potentially make a runtime borrow conflict harder to diagnose.